pallets/asset-index/src/lib.rs

```diff
@@ -525,13 +528,18 @@ pub mod pallet {
         /// Creates IndexAssetData if entry with given assetID does not exist.
         /// Otherwise adds the units to the existing holding
         fn add_asset(
+            metadata: AssetMetadata,
             asset_id: &T::AssetId,
             units: &T::Balance,
             availability: &AssetAvailability,
         ) -> DispatchResult {
             Holdings::<T>::try_mutate(asset_id, |value| -> Result<_, Error<T>> {
                 let index_asset_data = value.get_or_insert_with(|| {
-                    IndexAssetData::<T::Balance>::new(T::Balance::zero(), availability.clone())
+                    IndexAssetData::<T::Balance>::new(
+                        metadata,
```

Comment:
I think we can keep this separate, maybe introduce a designated storage map for metadata, like: https://github.com/paritytech/substrate/blob/master/frame/assets/src/lib.rs#L260